ext/mysqli/tests/gh15432.phpt

```diff
@@ -0,0 +1,24 @@
+--TEST--
+Bug GH-15432 (Heap corruption when querying a vector)
+--EXTENSIONS--
+mysqli
+--SKIPIF--
+<?php
+require 'connect.inc';
+$link = @my_mysqli_connect($host, $user, $passwd, $db, $port, $socket);
+if ($link === false) {
+    die(sprintf("skip Can't connect to MySQL Server - [%d] %s", mysqli_connect_errno(), mysqli_connect_error()));
+}
+if ($link->server_version < 90000) {
+    die("skip ads");
```

Comment:
```suggestion
    die("skip MySQL 9.0.0+ needed");
```
I realize it's not ideal that we have this skip section but I don't see any other choice. We might also restrict it against MariaDB since they are still working on this feature. When/if they add, the syntax may be different. I think I will add it in my PR too.

Reply:
Hmm, can we rely on `server_version` to be compatible for all supported servers (MySQL, MariaDB, Percona, TiDB, and what not)?

Reply:
Yes, we can. They agreed on such version numbers for this reason.